drivers/adc/adc_stm32.c

```diff
@@ -993,7 +999,8 @@ static int adc_stm32_init(const struct device *dev)
 	 * to be enabled.
 	 */
 	LL_ADC_SetCommonPathInternalCh(__LL_ADC_COMMON_INSTANCE(adc),
-				       LL_ADC_PATH_INTERNAL_VREFINT);
+				       LL_ADC_PATH_INTERNAL_VREFINT |
+				       LL_ADC_PATH_INTERNAL_VBAT);
```

Comment:
Why is this required? Isn't it handled by the `adc_stm32_setup_channels`?

Reply:
The one in the Init is required.
adc_stm32_setup_channels is not always called

Reply:
This looks unnecessary in `adc_stm32_init`, user should call `adc_stm32_channel_setup` before `adc_read` which will setup the `LL_ADC_PATH_INTERNAL_VREFINT` or `LL_ADC_PATH_INTERNAL_VBAT` accordingly.

Setting `LL_ADC_PATH_INTERNAL_VBAT` here will connect the VBAT pin to the internal voltage divider on init, this will take away the special feature of the VBAT pin (to save power), as now the voltage divider will drain the battery.

I recommend that `LL_ADC_SetCommonPathInternalCh` things to be removed from the init function. Based on my testing, removing them doesn't seem to cause any issue.